Source/Core/VideoCommon/VideoBackendBase.h

```diff
@@ -34,8 +39,8 @@ class VideoBackendBase
 {
 public:
   virtual ~VideoBackendBase() {}
-  virtual bool Initialize(const WindowSystemInfo& wsi) = 0;
-  virtual void Shutdown() = 0;
+  virtual bool Initialize(Core::System& system, const WindowSystemInfo& wsi) = 0;
+  virtual void Shutdown(Core::System& system) = 0;
```

Comment:
I suppose the reason you pass `System` to `Shutdown` instead of holding a pointer from `Initialize` is to prevent `VideoBackendBase` from having a dangling pointer member after `Shutdown`?

I'm thinking storing a `System*` might still be nicer than having all those other functions (e.g. `Video_GetQueryResult`) altered to also pass along the `System`.

Others can share their opinion, though.